src/VecSim/spaces/spaces.cpp

```diff
@@ -13,30 +13,26 @@ namespace spaces {
 void SetDistFunc(VecSimMetric metric, size_t dim, dist_func_t<float> *out_func,
                  unsigned char *alignment) {
 
-    static const Arch_Optimization arch_opt = getArchitectureOptimization();
-
     if (metric == VecSimMetric_Cosine || metric == VecSimMetric_IP) {
 
-        *out_func = IP_FP32_GetDistFunc(dim, arch_opt, alignment);
+        *out_func = IP_FP32_GetDistFunc(dim, nullptr, alignment);
 
     } else if (metric == VecSimMetric_L2) {
 
-        *out_func = L2_FP32_GetDistFunc(dim, arch_opt, alignment);
+        *out_func = L2_FP32_GetDistFunc(dim, nullptr, alignment);
     }
 }
 
 void SetDistFunc(VecSimMetric metric, size_t dim, dist_func_t<double> *out_func,
                  unsigned char *alignment) {
 
-    static const Arch_Optimization arch_opt = getArchitectureOptimization();
-
     if (metric == VecSimMetric_Cosine || metric == VecSimMetric_IP) {
 
-        *out_func = IP_FP64_GetDistFunc(dim, arch_opt, alignment);
+        *out_func = IP_FP64_GetDistFunc(dim, nullptr, alignment);
```

Comment:
why does codecov claim that its not covered? 🤔

Reply:
Not sure, we are running FP64 tests on coverage, right?